doc/connecting.md

```diff
@@ -176,3 +176,37 @@ df.to_sql("numbers", engine)
 %%sql
 SELECT * FROM numbers
 ```
+
+## Custom Connection
+
+```{versionadded} 0.7.1
+```
+
+If you are utilizing an engine that is not explicitly supported by SQLAlchemy, you can still leverage the JupySQL API through a customized connection.
+
+```{note}
+We currently support `%sql`, `%sqlplot`, and the `ggplot` API when using custom connection. However, please be advised that there may be some features/functionalities that won't be fully compatible with JupySQL.
+```
+
+For this example we'll use `QuestDB` which is not supported by SQLAlchemy.
+
+Initialize a connection, QuestDB works with the psycopg2 driver.
+
+```{note}
+Before running the following commands, please make sure you have QuestDB running locally. If you don't have it installed, you can install it via Docker by following the instructions [here](https://hub.docker.com/r/questdb/questdb).
+```
+
+```
+import psycopg2 as pg
```

Comment:
I'm not sure I completely understood. This DuckDB example allows me to run SQL on csv file without downloading it but in order to run the other examples we need to spin some docker containers (e.g custom connection and PostgreSQL).

Reply:
The problem with the example you have (psycopg2 + questdb) is that it isn't runnable, so if we break anything, we won't know because the code isn't running. What I'm proposing is to create an example that's runnable so if it breaks, we know.


So we can pick either sqlite or duckdb to make this example runnable, since they don't need a docker container running in the background. Perhaps the confusion comes from my example.

Currently, users can only do this:

```python
from sqlalchemy import create_engine
conn = create_engine("duckdb://")

%sql conn

```

But this will fail with the current version on pip:

```python
import duckdb
conn = duckdb.connect()

%sql conn
```

That's because the first example is a sqlalchemy connection but the second example is a generic driver. This PR allows the second example to work. So I'm suggesting adding it to the docs as a runnable example

~~~
```{code-cell} ipython3
import duckdb
conn = duckdb.connect()

%sql conn
```

~~~